rocAL/include/meta_data/meta_data.h

```diff
@@ -477,6 +510,81 @@ struct PolygonMaskBatch : public BoundingBoxBatch {
     std::vector<std::vector<std::vector<int>>> _vertices_counts = {};
 };
 
+class PixelwiseMaskBatch : public PolygonMaskBatch {
+   public:
+    void clear() override {
+        _bb_cords.clear();
+        _label_ids.clear();
+        _info_batch.clear();
+        _mask_cords.clear();
+        _polygon_counts.clear();
+        _vertices_counts.clear();
+        _buffer_size.clear();
+        _pixelwise_labels.clear();
+    }
+    MetaDataBatch& operator+=(MetaDataBatch& other) override {
+        _bb_cords.insert(_bb_cords.end(), other.get_bb_cords_batch().begin(), other.get_bb_cords_batch().end());
+        _label_ids.insert(_label_ids.end(), other.get_labels_batch().begin(), other.get_labels_batch().end());
+        _info_batch.insert(other.get_info_batch());
+        _mask_cords.insert(_mask_cords.end(), other.get_mask_cords_batch().begin(), other.get_mask_cords_batch().end());
+        _polygon_counts.insert(_polygon_counts.end(), other.get_mask_polygons_count_batch().begin(), other.get_mask_polygons_count_batch().end());
+        _vertices_counts.insert(_vertices_counts.end(), other.get_mask_vertices_count_batch().begin(), other.get_mask_vertices_count_batch().end());
+        _pixelwise_labels.insert(_pixelwise_labels.end(), other.get_pixelwise_labels_batch().begin(), other.get_pixelwise_labels_batch().end());
+        return *this;
+    }
+    void resize(int batch_size) override {
+        _bb_cords.resize(batch_size);
+        _label_ids.resize(batch_size);
+        _info_batch.resize(batch_size);
+        _mask_cords.resize(batch_size);
+        _polygon_counts.resize(batch_size);
+        _vertices_counts.resize(batch_size);
+        _pixelwise_labels.resize(batch_size);
+    }
+    std::vector<std::vector<int>>& get_pixelwise_labels_batch() override { return _pixelwise_labels; }
+    std::shared_ptr<MetaDataBatch> clone(bool copy_contents) override {
+        if (copy_contents) {
+            return std::make_shared<PixelwiseMaskBatch>(*this);
+        } else {
+            std::shared_ptr<MetaDataBatch> mask_batch_instance = std::make_shared<PixelwiseMaskBatch>();
+            mask_batch_instance->resize(this->size());
+            mask_batch_instance->get_info_batch() = this->get_info_batch();
+            return mask_batch_instance;
+        }
+    }
+    void copy_data(std::vector<void*> buffer) override {
+        if (buffer.size() < 2)
+            THROW("The buffers are insufficient")  // TODO -change
+        int* labels_buffer = (int*)buffer[0];
```

Comment:
Please use a static cast instead of c style casting
Applicable all such below instances in this function